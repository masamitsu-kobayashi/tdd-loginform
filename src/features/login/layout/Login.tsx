import React, { useEffect, useState } from "react";
import { SimpleButton } from "../elements/button/SimpleButton";
import { InputField } from "../elements/Input";
import { PasswordField } from "../elements/PasswordField";
import { UserIdFIeld } from "../elements/UserIdFIeld";
import {
  isInvalidEmail,
  isInvalidPasswordFormat,
  isUnderLength8,
} from "../utils/utils";

export const Login = () => {
  const [userId, setUserId] = useState<string>("");
  const handleChangeUserId = (e: React.ChangeEvent<HTMLInputElement>) => {
    setUserId(e.target.value);
  };

  const [password, setPassword] = useState<string>("");
  const handleChangePassword = (e: React.ChangeEvent<HTMLInputElement>) => {
    setPassword(e.target.value);
  };

  return (
    <>
      <UserIdFIeld value={userId} handleChange={handleChangeUserId} />
      <PasswordField value={password} handleChange={handleChangePassword} />
      <SimpleButton label="ログイン" />
    </>
  );
};
