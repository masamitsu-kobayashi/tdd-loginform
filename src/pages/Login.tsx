import React, { useState } from "react";
import { SimpleButton } from "../features/login/elements/button/SimpleButton";
import { PasswordField } from "../features/login/layout/PasswordField";
import { UserIdFIeld } from "../features/login/layout/UserIdFIeld";
import { loginRequest } from "../features/login/api/LoginApi";
import { useNavigate } from "react-router-dom";
import { useRecoilState } from "recoil";
import { userIdState } from "../atoms/atoms";
import {
  isInvalidEmail,
  isInvalidPasswordFormat,
} from "../features/login/utils/utils";

export const Login = () => {
  // const [userId, setUserId] = useState("");
  const [userId, setUserId] = useRecoilState(userIdState);
  const handleChangeUserId = (e: React.ChangeEvent<HTMLInputElement>) => {
    setUserId(e.target.value);
  };

  const [password, setPassword] = useState<string>("");
  const handleChangePassword = (e: React.ChangeEvent<HTMLInputElement>) => {
    setPassword(e.target.value);
  };

  const navigate = useNavigate();
  const handleClick = async () => {
    const { code } = await loginRequest(userId, password);
    if (code == "200") {
      navigate("/user");
    }
  };

  const isInvalidInput = () => {
    return isInvalidEmail(userId) || isInvalidPasswordFormat(password);
  };

  return (
    <>
      <UserIdFIeld value={userId} handleChange={handleChangeUserId} />
      <PasswordField value={password} handleChange={handleChangePassword} />
      <SimpleButton
        label="ログイン"
        handleClick={handleClick}
        disabled={isInvalidInput()}
      />
    </>
  );
};
