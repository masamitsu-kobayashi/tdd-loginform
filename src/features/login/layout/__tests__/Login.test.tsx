import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Login } from "../Login";

/*
  必要なコンポーネントが表示されている
  UserIDとパスワードが一致していたらOKとアラートが出る
  不一致なら、ユーザIDかパスワードに誤りがあります。と表示される
*/
describe("必要なコンポーネントが表示されている", () => {
  beforeEach(() => {
    render(<Login />);
  });

  it("スナップショットテスト", () => {
    const { asFragment } = render(<Login />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("user, Passw0rdと入力したら200,OKが帰ってくる", () => {
    const userIdFIeld = screen.getByPlaceholderText("UserId");
    const passwordFIeld = screen.getByPlaceholderText("password");
    const loginBtn = screen.getByRole("button");
    
  })


  // it("user, Passw0rdと入力したらアラートが出る", () => {

  // })

});
