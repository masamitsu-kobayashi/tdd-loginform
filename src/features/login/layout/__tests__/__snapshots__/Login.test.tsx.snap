// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`必要なコンポーネントが表示されている スナップショットテスト 1`] = `
<DocumentFragment>
  <div
    class="userId"
  >
    <input
      class="chakra-input css-0"
      placeholder="UserId"
      type="text"
      value=""
    />
    <label>
      メールアドレスを入力してください。
    </label>
  </div>
  <div
    class="password"
  >
    <input
      class="chakra-input css-0"
      placeholder="password"
      type="password"
      value=""
    />
    <label>
      8文字以上で入力してください。
    </label>
    <label>
      大文字小文字数字を少なくとも1つずつ以上使用してください。
    </label>
  </div>
  <button
    class="chakra-button css-4xx2wk"
    type="button"
  >
    ログイン
  </button>
</DocumentFragment>
`;
